src/services/saveImagesToUSB.ts

```diff
@@ -0,0 +1,49 @@
+import { IService } from '../types';
+import { readdirSync, existsSync, mkdirSync, rename, statSync, readdir } from 'fs';
+import { USB_WRITE_PATH } from 'config';
+
+export const SaveImagesToUSB: IService = {
+    execute: async () => {
+        try {
+            //Check if USB is connected
+            const usbConnected = existsSync(USB_WRITE_PATH);
+            if (usbConnected) {
+                const directoryPath = USB_WRITE_PATH;
+                const files = readdir(directoryPath, (err, files) => {
+                    files.forEach((file) => {
```

Comment:
Correct me if I'm wrong, but forEach will execute all of them pretty much at the same time.
Not sure if it's healthy for CPU or not,

I'd do a regular for-loop and promisify `rename` non-blocking function,
(you can take a look on usage of `promisify` in project),
and then do 
for () {
await promisifiedRename...
}

this way for-loop + await should still be non-blocking, but you will also execute those operations one-by-one, not at the same time.